interfaces/repo.go

```diff
@@ -579,8 +579,8 @@ func (r *Repository) Connect(ref ConnRef) error {
 	}
 
 	// TODO: store copy of attributes from hooks
-	cplug := &ConnectedPlug{plugInfo: plug}
-	cslot := &ConnectedSlot{slotInfo: slot}
+	cplug := NewConnectedPlug(plug, nil)
+	cslot := NewConnectedSlot(slot, nil)
```

Comment:
Doesn't feel like that second argument is relevant. If you don't yet have a use case in other branches, can we please drop it?

Reply:
The second argument is for passing dynamic attributes when constructing connected plug/slot. It's not used in this branch but it is the core of the feature and is used in the followup integration branch.